src/pgtz/build.rs

```diff
@@ -130,42 +140,103 @@ fn main() -> Result<()> {
-                Err(_) => {
-                    // Link to timezone rather than fixed offset from UTC. These
-                    // hard to handle as the offset from UTC changes depending on
-                    // the current date/time. Skip for now.
-                    continue;
-                }
+            let spec = match pieces[1].parse::<i32>() {
+                Ok(utc_offset_secs) => TimezoneAbbrevSpec::FixedOffset {
+                    utc_offset_secs,
+                    is_dst: pieces.get(2) == Some(&"D"),
+                },
+                Err(_) => TimezoneAbbrevSpec::Tz(pieces[1]),
             };
-            let is_dst = pieces.get(2) == Some(&"D");
 
             rust_buf.write_block(
                 format!("pub const {abbrev}: TimezoneAbbrev = TimezoneAbbrev"),
                 |rust_buf| {
-                    rust_buf.writeln(format!("abbrev: \"{abbrev}\","));
-                    rust_buf.writeln(format!("utc_offset_secs: {utc_offset_secs},"));
-                    rust_buf.writeln(format!("is_dst: {is_dst}"));
+                    rust_buf.writeln(format!("abbrev: {},", abbrev.quoted()));
+                    match &spec {
+                        TimezoneAbbrevSpec::FixedOffset {
+                            utc_offset_secs,
+                            is_dst,
+                        } => {
+                            rust_buf.write_block(
+                                "spec: TimezoneAbbrevSpec::FixedOffset",
+                                |rust_buf| {
+                                    rust_buf.writeln(format!(
+                                        "offset: make_fixed_offset({utc_offset_secs}),"
+                                    ));
+                                    rust_buf.writeln(format!("is_dst: {is_dst},"));
+                                },
+                            );
+                        }
+                        TimezoneAbbrevSpec::Tz(name) => {
+                            let name = name.replace('/', "__");
+                            rust_buf.writeln(format!("spec: TimezoneAbbrevSpec::Tz(Tz::{name})"));
+                        }
+                    }
                 },
             );
             rust_buf.writeln(";");
 
+            let (sql_utc_offset, sql_is_dst) = match &spec {
+                TimezoneAbbrevSpec::FixedOffset {
+                    utc_offset_secs,
+                    is_dst,
+                } => {
+                    let utc_offset = format!("interval '{utc_offset_secs} seconds'");
+                    let is_dst = is_dst.to_string();
+                    (utc_offset, is_dst)
+                }
+                TimezoneAbbrevSpec::Tz(name) => {
+                    let utc_offset = format!("timezone_offset('{name}', now()).base_utc_offset + timezone_offset('{name}', now()).dst_offset");
+                    let is_dst =
+                        format!("timezone_offset('{name}', now()).dst_offset > interval '0'");
```

Comment:
Can this ever be negative? Could it ever be? Safer to do `<>` instead?

Reply:
Oh great point, yes, thanks. Will fix.

Reply:
Fixed.